### PR TITLE
Fix filtering directories on compression extension match

### PR DESCRIPTION
**Pull request by tgravescs**
If you have a directory with a "." in it, then it matches on our filter to check for filename extensions for compressed files.

Change the checks to make sure to only look at file name extensions when its a file.

Note this was caught by our integration tests so no test added here.

Signed-off-by: Thomas Graves <tgraves@apache.org>

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM, 
  - consider a unit test for the regression

File: tools/src/main/scala/com/nvidia/spark/rapids/tool/EventLogPathProcessor.scala [hunk 1]
```diff
@@ -112,14 +112,15 @@ object EventLogPathProcessor extends Logging {
         val (validLogs, invalidLogs) = fs.listStatus(inputPath).partition(s => {
             val name = s.getPath().getName()
             (s.isFile ||
-              (s.isDirectory &&
-                (isEventLogDir(name) || isDatabricksEventLogDir(s, fs))))
+              (s.isDirectory && (isEventLogDir(name) || isDatabricksEventLogDir(s, fs))))
```
Comment (gerashegalov) [inline tools/src/main/scala/com/nvidia/spark/rapids/tool/EventLogPathProcessor.scala:115]: looks like a purely cosmetic change that we could keep out of the bug fix.

**Comment by tgravescs** [comment]
> Note this was caught by our integration tests so no test added here.

**Pull request merged**